loader/icd.h

```diff
@@ -33,6 +33,18 @@
 #define CL_USE_DEPRECATED_OPENCL_1_2_APIS
 #endif
 
+#ifndef CL_USE_DEPRECATED_OPENCL_2_0_APIS
+#define CL_USE_DEPRECATED_OPENCL_2_0_APIS
+#endif
+
+#ifndef CL_USE_DEPRECATED_OPENCL_2_1_APIS
+#define CL_USE_DEPRECATED_OPENCL_2_1_APIS
+#endif
+
+#ifndef CL_USE_DEPRECATED_OPENCL_2_2_APIS
+#define CL_USE_DEPRECATED_OPENCL_2_2_APIS
+#endif
+
```

Comment:
Note: We're currently including the OpenCL headers in two places in this project: in icd.h here and in icd_dispatch.h.  As a further improvement it would be nice to reduce this duplication.  Things we could consider are:

1. Have icd_dispatch.h include icd.h to pick up the headers.  This is basically already done in practice, because we're always including icd.h and icd_dispatch.h as a pair.
2. Move the dispatch table structures into icd.h and remove icd_dispatch.h.
3. Extract the OpenCL header includes into a third file and include it from both icd.h and icd_dispatch.h.

Reply:
@bashbaug Do you want to address this before we merge, or should I go ahead?